mlflow/tracing/processor/inference_table.py

```diff
@@ -51,10 +52,23 @@ def on_start(self, span: OTelSpan, parent_context: Optional[Context] = None):
                 span is obtained from the global context, it won't be passed here so we should not
                 rely on it.
         """
-        request_id = _get_flask_request().headers.get(_HEADER_REQUEST_ID_KEY)
-        if not request_id:
-            _logger.debug("Request ID not found in the request headers. Skipping trace processing.")
-            return
+        request_id = maybe_get_request_id()
+        if request_id is None:
+            # If this is invoked outside of a flask request, it raises error about
+            # outside of request context. We should avoid this by skipping the trace processing
```

Comment:
can we use https://flask.palletsprojects.com/en/2.3.x/api/#flask.has_request_context instead of try-catch?